src/test/java/hudson/security/LdapMultiEmbeddedTest.java

```diff
@@ -135,4 +146,153 @@ public void loginWithBrokenServerInTheMiddle() throws Exception {
+                break;
+            }
+        }
+        assertNotNull(repl);
+        assertTrue(index >= 0);
+
+        LDAPConfiguration nc = new LDAPConfiguration(
+                repl.getServer(),
+                repl.getRootDN(),
+                true,
+                repl.getManagerDN(),
+                Secret.fromString("something completely wrong"));
+        configurations.set(index, nc);
+        r.jenkins.setSecurityRealm(new LDAPSecurityRealm(configurations,
+                realm.disableMailAddressResolver,
+                realm.getCache(),
+                realm.getUserIdStrategy(),
+                realm.getGroupIdStrategy()));
+    }
+
+    @Test
+    public void when_first_is_wrong_and_login_on_first_then_log() throws Exception {
+        setBadPwd(planetExpress);
+        try {
+            r.createWebClient().login("fry", "fry");
+            fail("Login should fail");
+        } catch (FailingHttpStatusCodeException e) {
+            assertEquals(500, e.getStatusCode());
+        }
+        assertThat(log, recorded(Level.WARNING,
+                allOf(containsString(FAILED_COMMUNICATION_WITH_LDAP_SERVER),
+                        containsString(planetExpress.getUrl())),
+                CoreMatchers.<Throwable>instanceOf(DataAccessException.class)));
+    }
+
+    @Test
+    public void when_first_is_wrong_and_login_on_second_then_log() throws Exception {
+        setBadPwd(planetExpress);
+        try {
+            r.createWebClient().login("hnelson", "pass");
+            fail("Login should fail");
+        } catch (FailingHttpStatusCodeException e) {
+            assertEquals(500, e.getStatusCode());
+        }
+        assertThat(log, recorded(Level.WARNING,
+                allOf(containsString(FAILED_COMMUNICATION_WITH_LDAP_SERVER),
+                        containsString(planetExpress.getUrl())),
+                CoreMatchers.<Throwable>instanceOf(DataAccessException.class)));
+    }
+
+    @Test
+    public void when_second_is_wrong_and_login_on_second_then_log() throws Exception {
+        setBadPwd(sevenSeas);
+        try {
+            r.createWebClient().login("hnelson", "pass");
+            fail("Login should fail");
+        } catch (FailingHttpStatusCodeException e) {
+            assertEquals(500, e.getStatusCode());
+        }
+        assertThat(log, recorded(Level.WARNING,
+                allOf(containsString(FAILED_COMMUNICATION_WITH_LDAP_SERVER),
+                        containsString(sevenSeas.getUrl())),
+                CoreMatchers.<Throwable>instanceOf(DataAccessException.class)));
+    }
+
+    @Test
+    public void when_second_is_wrong_and_login_on_first_no_log() throws Exception {
+        setBadPwd(sevenSeas);
+        r.createWebClient().login("fry", "fry");
+
+        assertThat(log, not(recorded(Level.WARNING,
+                containsString(FAILED_COMMUNICATION_WITH_LDAP_SERVER),
+                CoreMatchers.<Throwable>instanceOf(DataAccessException.class))));
```

Comment:
No need to match the exception if we are anyway asserting that there is no such warning, right?